src/rviz/TrajectoryMarker.cpp

```diff
@@ -0,0 +1,259 @@
+{
+  mMarkerServer->erase(mInteractiveMarker.name);
+}
+
+//==============================================================================
+void TrajectoryMarker::setTrajectory(trajectory::ConstTrajectoryPtr trajectory)
+{
+  using visualization_msgs::Marker;
+
+  if (trajectory)
+  {
+    auto statespace = trajectory->getStateSpace();
+    if (!std::dynamic_pointer_cast<statespace::dart::MetaSkeletonStateSpace>(
+            statespace))
+    {
+      throw std::invalid_argument(
+          "The statespace in the trajectory should be MetaSkeletonStateSpace");
+    }
+  }
+
+  mTrajectory = std::move(trajectory);
+
+  mNeedPointsUpdate = true;
+}
+
+//==============================================================================
+trajectory::ConstTrajectoryPtr TrajectoryMarker::getTrajectory() const
+{
+  return mTrajectory;
+}
+
+//==============================================================================
+void TrajectoryMarker::setColor(const Eigen::Vector3d& rgb)
+{
+  auto& marker = getMarker();
+  marker.color.r = rgb[0];
+  marker.color.g = rgb[1];
+  marker.color.b = rgb[2];
+
+  mNeedUpdate = true;
+}
+
+//==============================================================================
+Eigen::Vector3d TrajectoryMarker::getColor() const
+{
+  const auto& marker = getMarker();
+  return convertROSColorRGBAToEigen(marker.color).head<3>();
+}
+
+//==============================================================================
+void TrajectoryMarker::setAlpha(double alpha)
+{
+  auto& marker = getMarker();
+  marker.color.a = alpha;
+
+  mNeedUpdate = true;
+}
+
+//==============================================================================
+double TrajectoryMarker::getAlpha() const
+{
+  const auto& marker = getMarker();
+  return marker.color.a;
+}
+
+//==============================================================================
+void TrajectoryMarker::setRGBA(const Eigen::Vector4d& rgba)
+{
+  auto& marker = getMarker();
+  marker.color = convertEigenToROSColorRGBA(rgba);
+
+  mNeedUpdate = true;
+}
+
+//==============================================================================
+Eigen::Vector4d TrajectoryMarker::getRBGA() const
+{
+  const auto& marker = getMarker();
+  return convertROSColorRGBAToEigen(marker.color);
+}
+
+//==============================================================================
+void TrajectoryMarker::setThickness(double thickness)
+{
+  auto& marker = getMarker();
+  marker.scale.x = thickness;
+
+  mNeedUpdate = true;
+}
+
+//==============================================================================
+double TrajectoryMarker::getThickness() const
+{
+  const auto& marker = getMarker();
+  return marker.scale.x;
+}
+
+//==============================================================================
+void TrajectoryMarker::setNumLineSegments(std::size_t numLineSegments)
+{
+  if (numLineSegments < 2)
+  {
+    throw std::invalid_argument(
+        "Number of line segments should be greater than 2");
```

Comment:
This error message doesn't match the check -- did you mean "greater than or equal to 2"?

Either way, I'm not sure if this should be an actual error that we throw since it only means that the visualized trajectory won't look very good.

Reply:
Treating this as not an error is fine to me. In that case, I would like to at least (1) print a warning so that the user doesn't get confused why the trajectory is not visualized and (2) throw exception when `numLineSegments` is negative.